src/syft/core/node/common/action/run_class_method_action.py

```diff
@@ -85,9 +86,25 @@ def execute_action(self, node: AbstractNode, verify_key: VerifyKey) -> None:
 
         result = method(resolved_self.data, *resolved_args, **resolved_kwargs)
 
-        # ensure that result has the right id
-        # TODO: overload all methods to incorporate this automatically
-        result.id = self.id_at_location
+        if isprimitive(value=result):
+            # Wrap in a PythonPrimitive
+            primitive = PythonPrimitive(value=result, id=self.id_at_location)
+            result = StorableObject(
+                id=self.id_at_location,
+                data=primitive,
+                read_permissions=result_read_permissions,
+            )
+        else:
+            # TODO: overload all methods to incorporate this automatically
+            if hasattr(result, "id"):
```

Comment:
Q: At this point we should have an id for all the results, right?
I think the best approach will be to throw an exception since not having an id is something that we did not expect

Reply:
So, it turns out that one of the tests fails on the else of that code, but I have commented out the exception I added.

Reply:
Try uncommenting the ```
raise Exception(f"Result has no ID. {result}")
```

I have to finish for the day so I didnt get time to investigate whats going on but something is going on here.

Reply:
agreed - we should throw an error here. good job